conftest.py

```diff
@@ -54,82 +75,73 @@ def pytest_addoption(parser: Any) -> None:
                      default='bitcoinrpc',
                      help="the RPC username for your test bitcoin instance (default=bitcoinrpc)")
     parser.addoption("--nirc",
-                         type="int",
-                         action="store",
-                         default=1,
-    help="the number of local miniircd instances")
-
-def teardown() -> None:
-    #didn't find a stop command in miniircd, so just kill
-    global miniircd_procs
-    for m in miniircd_procs:
-        m.kill()
-    #shut down bitcoin and remove the regtest dir
-    local_command([os.path.join(bitcoin_path, "bitcoin-cli"), "-regtest",
-                    "-rpcuser=" + bitcoin_rpcusername,
-                   "-rpcpassword=" + bitcoin_rpcpassword, "stop"])
-    #note, it is better to clean out ~/.bitcoin/regtest but too
-    #dangerous to automate it here perhaps
+                     type="int",
+                     action="store",
+                     default=1,
+                     help="the number of local miniircd instances")
 
 
 @pytest.fixture(scope="session", autouse=True)
-def setup(request) -> None:
-    request.addfinalizer(teardown)
-
-    global bitcoin_conf, bitcoin_path, bitcoin_rpcpassword, bitcoin_rpcusername
-    bitcoin_path = request.config.getoption("--btcroot")
-    bitcoin_conf = request.config.getoption("--btcconf")
-    print("Here is the bitcoin_conf path:")
-    print(bitcoin_conf)
-    bitcoin_rpcpassword = request.config.getoption("--btcpwd")
-    bitcoin_rpcusername = request.config.getoption("--btcuser")
-
-    #start up miniircd
-    #minor bug in miniircd (seems); need *full* unqualified path for motd file
+def miniircd(pytestconfig):
```

Comment:
I think this would be better function name.
```suggestion
def setup_miniircd(pytestconfig):
```

Reply:
Yeah, makes sense. My idea was that if we want, one day, to have a return value (for example the bitcoind `subprocess` object), then IMHO it's better to have `def test_foo(bitcoind)` than `def test_foo(setup_bitcoind)`.

But given that we don't currently do that, and that we can always rename it later in case, I will change it.